addons/hr_holidays/models/hr_leave.py

```diff
@@ -349,13 +349,12 @@ def _inverse_description(self):
 
     def _search_description(self, operator, value):
         is_officer = self.user_has_groups('hr_holidays.group_hr_holidays_user')
-        domain = [('private_name', operator, value)]
+        domain = [('private_name', operator, value), ('company_id', 'in', self.env.companies.ids)]
```

Comment:
Why do we need to add this? :thinking:

Reply:
If you have 10 companies in DB, and if a user in only one company make a search _search make a query for all companies. It is un optimization, but we have remove it.

Reply:
When you fetch, search, we apply ir_rules, and for hr.leave we do have multi-company rule, so it will be checked regardless.